src/model/client.ts

```diff
@@ -246,9 +340,14 @@ export class CommandoClient extends Client {
 
     const command = Command.find(this, commandName)
 
-    if (!command) {
+    if (!command || command.options.unknown || (command.options.guildOnly && msg.guild)) {
       this.emit('invalidCommand', msg)
```

Comment:
I'm surprised I didnt catch this earlier...

These should be static typed, not magic strings.

This should either be an enum or a type. I'm leaning more towards an enum.

I'd recommend to create a `Event` enum that holds all of our custom events.

```js
export enum Event {
  INVALID_COMMAND = 'invalidCommand',
  ...
}
```

Then you would change this line to `this.emit(Event.INVALID_COMMAND, msg)`